### ISSUE TITLE
Can't use typed array (Uint8, Uint32, etc)

### ISSUE THREAD

**Issue by ShieldCoin** [gpujs/gpu.js#332]
When declaring a typed array like so:
```
let a = new Uint32Array([1, 2, 3])
```
I get the error:
```
gpu.js:1484:4
Unknown ast type : NewExpression
```
Suggesting that you can't create a new typed array class.
I've looked through the documentation but can't find anything on typed arrays.
I found #41, but no explanation of how typed arrays are supposed to be used in gpu.js
Are there any plans to implement typed arrays, or is there already a workaround for it?

**Comment by robertleeplummerjr** [comment]
Can you show more source for context?

**Comment by ShieldCoin** [comment]
```
 gpu.createKernel(function () {
    var array = Uint32Array(10);
    for (var i = 0; i < 10; i++) {
      array[i] += this.thread.x;
    }
    return array;
  }).setOutput([10])
```

**Comment by robertleeplummerjr** [comment]
TL;DR
In short, this isn't compatible on the gpu size, and thus we've built the environment to fail in both.  This "limitation" is for the good of performance, but that doesn't mean you cannot achieve what you are trying to do.  That being said: what is it you are trying to do?


TS;DR

Lets break down the usage of the kernel you are creating.  In the first example you do this:
```js
var array = Uint32Array(10);
```

While there are some means to accomplish this on the GPU, this example isn't really compatible with GPU architectures.  However, gpu.js builds the array from the thread.  You have to _think of the kernel as the loop_.   A compatible kernel would be something like this:

```js
gpu.createKernel(function () {
  return this.thread.x;  
}).setOutput([10, 10])
```

The second example could be done using constants, and those should allow you to use typed arrays.  An example to possibly support that use case:

```js
gpu.createKernel(function (array) {
  return this.constants.a[this.thread.x];
})
  .setOutput([10, 3])
  .setConstants({
    a: [1,1,1]
  });
```

In either case, I think we could still get your use case supported if we had more context.

**Comment by ShieldCoin** [comment]
```
    const IV = UInt32Array([241421,5167192,512954])
    var h = IV
    var a = [MAX_LENGTH & 0xff, 0, 1, 1]
    h[0] = (a[8 + 0] & 0xff) | ((a[8 + 1] & 0xff) << 8) | ((a[8 + 2] & 0xff) << 16) | ((a[8 + 3] & 0xff) << 24)
    var x = UInt8Array(MAXB_LENGTH)
..... Some code modifying h again
     for (i = 0; i < left; i++) {
       x[nx + i] = input[offset + i]
     }
    var d = UInt8Array(32)
    for (i = 0; i < 8; i++) {
      var n = h[i]
      d[i * 4 + 0] = (n >>> 0) & 0xff
      d[i * 4 + 1] = (n >>> 8) & 0xff
      d[i * 4 + 2] = (n >>> 16) & 0xff
      d[i * 4 + 3] = (n >>> 24) & 0xff
    }
```
This is about the context for what I'm trying to do, it requires  UInt8Array (or something similar with the same functionality) to properly execute.

If I can get a regular UInt32Array I can probably make a function to update it UInt8Array specifications when changes are applied.

Constants do seem interesting I can get IV out of the function, which is good, but it would really help if I was to be able to assign values to arrays

**Comment by robertleeplummerjr** [comment]
I want to clarify, that you can achieve what you desire, by simply assigning a constant and reusing it inside the kernel.  But the exact _declaration of the an array within the kernel_ is what isn't possible.
